packages/mdc-top-app-bar/foundation.ts

```diff
@@ -50,64 +50,23 @@ export class MDCTopAppBarBaseFoundation extends MDCFoundation<MDCTopAppBarAdapte
       hasClass: () => false,
       setStyle: () => undefined,
       getTopAppBarHeight: () => 0,
-      registerNavigationIconInteractionHandler: () => undefined,
-      deregisterNavigationIconInteractionHandler: () => undefined,
       notifyNavigationIconClicked: () => undefined,
-      registerScrollHandler: () => undefined,
-      deregisterScrollHandler: () => undefined,
-      registerResizeHandler: () => undefined,
-      deregisterResizeHandler: () => undefined,
       getViewportScrollY: () => 0,
       getTotalActionItems: () => 0,
     };
     // tslint:enable:object-literal-sort-keys
   }
 
-  protected scrollHandler_?: SpecificEventListener<'scroll'>;
-  protected resizeHandler_?: SpecificEventListener<'resize'>;
-  private readonly navClickHandler_: SpecificEventListener<'click'>;
+  handleScroll?: SpecificEventListener<'scroll'>;
```

Comment:
Ideally these methods should be overriden in respective child classes of TopAppBar foundation.

In current model each child foundation of TopAppBar is overwriting this property by reassigning new function, but each foundation should override these functions instead.